### PR TITLE
[Merged by Bors] - feat(Data/Finset/NatDivisors): Nat.divisors of a product

### PR DESCRIPTION
**Pull request by adomani**
The function `Nat.divisors` as a multiplicative homomorphism `ℕ →* Finset ℕ`.

This result is in a new file `Data/Finset/NatDivisors`, to avoid adding imports and/or lengthening existing files.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by urkud** [comment]
I'm not sure that you need more imports for the new proof.

**Comment by adomani** [comment]
Yuri, I added the Finset and the List version.  I have to go now, but I'll look into the Multiset version and whether this fits better in an earlier file later!

Thanks!

File: Mathlib/Data/Finset/NatDivisors.lean [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+lemma Prime.divisors_sq {p : ℕ} (hp : p.Prime) : (p ^ 2).divisors = {p ^ 2, p, 1} := by
+  simp [divisors_prime_pow hp, range_succ]
+
+end Nat
+
+open Nat
+
+lemma List.nat_divisors_prod (l : List ℕ) : divisors l.prod = (l.map divisors).prod := by
+  induction l with
+  | nil => simp; rfl
```
Comment (urkud) [inline Mathlib/Data/Finset/NatDivisors.lean:34]: Why do you need a non-terminal `simp` here?

Comment (adomani) [reply Mathlib/Data/Finset/NatDivisors.lean:34]: The goal at the `rfl` is `⊢ {1} = 1`.

**Comment by adomani** [comment]
Yuri, I added the `Finset` lemmas, but either I took a wrong turn, or there was some missing API around `Pointwise`.

Anyway, I could still not move the lemmas in a different file, since I could not find any interaction between `Nat.divisors` and `Pointwise`: either a file develops the theory of `Nat.divisors` and has no `Pointwise`, or there is `Pointwise`, but no `Nat.divisors`.

From this point of view, it may make sense to create this new file, what do you think?

**Comment by adomani** [comment]
Yuri, your latest commit is *such an improvement* over my butchering of the lemmas!  Thanks!

**Comment by urkud** [comment]
Thanks! :tada:
bors merge

**Pull request closed without merging**